### DISCUSSION TITLE
ExtractStateValue

### DISCUSSION THREAD

**Discussion by AliKholmatov** [statelyai/xstate#3489] [General]
![image](https://user-images.githubusercontent.com/87648847/182158138-d64cbe55-bc53-4e67-9284-11a0a833ad4f.png)

in latest versions ExtractStateValue was deleted. What can I use instead of ExtractStateValue?

**Comment by Andarist** [comment]
Could you describe your issue? Why do you need this type? Why do you want to declare a state schema up front? We plan to remove this generic in the upcoming major version of XState so it would be great to get a better understanding why people are still using it in v4
